File: 01. SourceCode/IOS/swipesafe/swipesafe/Models/Report/ChildModel.swift
//
//  ChildModel.swift
//  swipesafe
//
//  Created by Thanh Luu on 3/17/19.
//  Copyright © 2019 childfund. All rights reserved.
//

import Foundation

class ChildModel: Encodable {
    var id: String = ""
    var reportId: String = ""
    var name: String = ""
    var gender: String = ""
    var age: Int?
    var birthDay: String = ""
    var level: String = ""
    var address: String = ""
    var provinceId: String = ""
    var districtId: String = ""
    var wardId: String = ""
    var fullAddress: String = ""
    var dateAction: String = ""
    var abuses: [ChildAbuseModel] = []
    
    var genderName: String = ""
    var levelName: String = ""
    var provinceName: String = ""
    var districtName: String = ""
    var wardName: String = ""
    
    enum CodingKeys: String, CodingKey {
        case id = "Id"
        case reportId = "ReportId"
        case name = "Name"
        case gender = "Gender"
        case age = "Age"
        case birthDay = "BirthDay"
        case level = "Level"
        case address = "Address"
        case provinceId = "ProvinceId"
        case districtId = "DistrictId"
        case wardId = "WardId"
        case fullAddress = "FullAddress"
        case dateAction = "DateAction"
        case abuses = "ListAbuse"
    }
}

File: 01. SourceCode/ChildFund/SwipeSafe.Model/SwipeSafeModel/ChildModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SwipeSafe.Model.SwipeSafeModel
{
    public class ChildModel
    {
        public string Id { get; set; }
        public string ReportId { get; set; }
        public string Name { get; set; }
        public string Gender { get; set; }
        public int? Age { get; set; }
        public Nullable<System.DateTime> Birthday { get; set; }
        public string Level { get; set; }
        public string Address { get; set; }
        public string ProvinceId { get; set; }
        public string DistrictId { get; set; }
        public string WardId { get; set; }
        public string FullAddress { get; set; }
        public Nullable<System.DateTime> DateAction { get; set; }
        
        /// <summary>
        /// Hành vi xam hại
        /// </summary>
        public List<ChildAbuseModel> ListAbuse { get; set; }
    }
}
